TMG.EMME/TMGToolbox2/src/Assign/assign_boarding_penalty.py

```diff
@@ -0,0 +1,347 @@
+# ---LICENSE----------------------
+"""
+    Copyright 2015 Travel Modelling Group, Department of Civil Engineering, University of Toronto
+
+    This file is part of the TMG Toolbox.
+
+    The TMG Toolbox is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The TMG Toolbox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with the TMG Toolbox.  If not, see <http://www.gnu.org/licenses/>.
+"""
+# ---METADATA---------------------
+"""
+Assign V4 Boarding Penalties
+
+    Authors: pkucirek
+
+    Latest revision by: James Vaughan
+    
+    
+    Assigns line-specific boarding penalties (stored in UT3) based on specified
+    groupings, for transit assignment estimation.
+        
+"""
+# ---VERSION HISTORY
+"""
+    0.0.1 Created on 2014-02-14 by TrajceNikolov
+    
+    1.0.0 Cleaned and published. Now accepts a list of scenarios, for easy use (Modeller only).
+            Also changed the order of parameters to better match the order of groups used in
+            GTAModel V4.
+            
+    1.0.1 Added short description
+
+    1.1.0 Line groups are no longer hard-coded. Instead, user-inputted selector expressions are used
+            and the number of line groups is open-ended.
+
+    1.2.0 Added ability to set IVTT perception factor.
+
+    1.2.1 Added ability to set Transfer boarding penalties.
+
+"""
```

Comment:
This should be updated for TMG Toolbox 2 for EMME and taking notes of any revisions that you have done.